src/Microsoft.IdentityModel.Tokens.Saml/Saml2/Saml2SecurityTokenHandler.cs

```diff
@@ -500,6 +525,35 @@ internal static bool IsSaml2Assertion(XmlReader reader)
             return reader.IsStartElement(Saml2Constants.Elements.Assertion, Saml2Constants.Namespace);
         }
 
+        /// <summary>
+        /// Indicates if the assertion string represents an EncryptedAssertion.
+        /// </summary>
+        /// <param name="assertion">String representation of an assertion.</param>
+        /// <returns>'true' if assertion is encrypted, 'false' otherwise.</returns>
+        internal static bool IsSaml2EncryptedAssertion(string assertion)
+        {
+            using (var stringReader = new StringReader(assertion))
+            {
+                var settings = new XmlReaderSettings { DtdProcessing = DtdProcessing.Prohibit };
+#if NET45 || NET451
+                settings.XmlResolver = null;
+#endif
+                using (var reader = XmlReader.Create(stringReader, settings))
```

Comment:
>XmlReader.Create(stringReader, settings)) [](start = 36, length = 41)

Suggestion: The library could have a utility function to create this one ensuring that everyone creates the XmlReader with the appropriate (secure) XmlReaderSettings

Reply:
agree - same settings are created more than once throughout the project.

Reply:
addressed with: bfaa792c407991dd698f5922841e4bfa03884662